be/src/exprs/string_functions.cpp

```diff
@@ -2109,31 +2109,47 @@ DEFINE_STRING_UNARY_FN_WITH_IMPL(url_decodeImpl, str) {
     return StringFunctions::url_decode_func(str.to_string());
 }
 
-std::string StringFunctions::url_decode_func(const std::string& value) {
-    std::string ret;
-    char ch;
-    int ii;
-    char l, r;
-    size_t length = value.length();
+static Status url_decode_slice(const Slice& src, Slice* to) {
+    size_t length = src.size;
+    auto* value = src.data;
+    char* p = to->data;
     for (size_t i = 0; i < length; i++) {
         if (value[i] == '%') {
-            l = value[i + 1];
-            r = value[i + 2];
+            char l = value[i + 1];
+            char r = value[i + 2];
             if ((l < 'A' || l > 'F') && (l < '0' || l > '9')) {
-                throw std::runtime_error("decode string contains illegal hex chars: " + value.substr(i + 1, 2));
+                return Status::RuntimeError(
+                        strings::Substitute("decode string contains illegal hex chars: $0$1", l, r));
             }
             if ((r < 'A' || r > 'F') && (r < '0' || r > '9')) {
-                throw std::runtime_error("decode string contains illegal hex chars: " + value.substr(i + 1, 2));
+                return Status::RuntimeError(
+                        strings::Substitute("decode string contains illegal hex chars: $0$1", l, r));
             }
-            sscanf(value.substr(i + 1, 2).c_str(), "%x", &ii);
-            ch = static_cast<char>(ii);
-            ret += ch;
+            auto mask = (l - 'A') >> 8;
```

Comment:
need more explain

Reply:
added

Reply:
no need change `url_encode_func `?  to keep the opposite actions with this decode?